collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -594,46 +597,22 @@ private void loadFromIntent(Intent intent) {
             }
 
             instancePath = formInfo.getInstancePath();
+            List<Form> candidateForms = formsRepository.getAll(formInfo.getFormId(), formInfo.getFormVersion());
 
-            String jrFormId = formInfo.getFormId();
-            String jrVersion = formInfo.getFormVersion();
-
-            String[] selectionArgs;
-            String selection;
-            if (jrVersion == null) {
-                selectionArgs = new String[]{jrFormId};
-                selection = FormsColumns.JR_FORM_ID + "=? AND "
-                        + FormsColumns.JR_VERSION + " IS NULL";
-            } else {
-                selectionArgs = new String[]{jrFormId, jrVersion};
-                selection = FormsColumns.JR_FORM_ID + "=? AND "
-                        + FormsColumns.JR_VERSION + "=?";
-            }
-
-            int formCount = FormsDaoHelper.getFormsCount(selection, selectionArgs);
-            if (formCount < 1) {
+            if (candidateForms.isEmpty()) {
                 createErrorDialog(getString(
                         R.string.parent_form_not_present,
-                        jrFormId)
-                                + ((jrVersion == null) ? ""
-                                : "\n"
-                                + getString(R.string.version)
-                                + " "
-                                + jrVersion),
+                        formInfo.getFormId())
+                                + ((formInfo.getFormVersion() == null) ? ""
+                                : "\n" + getString(R.string.version) + " " + formInfo.getFormVersion()),
                         EXIT);
                 return;
-            } else {
-                formPath = FormsDaoHelper.getFormPath(selection, selectionArgs);
-
-                /**
-                 * Still take the first entry, but warn that there are multiple rows. User will
```

Comment:
This just wasn't true. The dialog errored out and quit the form so this wasn't a warning and the first entry was never used. Also it's no longer true that the SQLite database needs to be touched directly.